azure-servicebus/src/main/java/com/microsoft/azure/servicebus/MessageReceiver.java

```diff
@@ -679,9 +678,19 @@ public Instant renewMessageLock(IMessage message) throws InterruptedException, S
         return Utils.completeFuture(this.renewMessageLockAsync(message));
     }
 
+    @Override
+    public Instant renewMessageLock(UUID lockToken)  throws InterruptedException, ServiceBusException {
+        return Utils.completeFuture(this.renewMessageLockAsync(lockToken));
+    }
+
     //	@Override
     public Collection<Instant> renewMessageLockBatch(Collection<? extends IMessage> messages) throws InterruptedException, ServiceBusException {
-        return Utils.completeFuture(this.renewMessageLockBatchAsync(messages));
+        Collection<Instant> newLockedUntilUtcTimes = new ArrayList<>(messages.size());
+        for(IMessage message : messages) {
+            newLockedUntilUtcTimes.add(this.renewMessageLock(message));
+        }
```

Comment:
Don't call renewMessageLock in a for loop. Just put all lock tokens in an array and call renewMessageLockBatchAsync(UUID[] lockTokens).. That will be one call to the service.